jib-core/src/integration-test/java/com/google/cloud/tools/jib/builder/BuildStepsIntegrationTest.java

```diff
@@ -148,14 +150,16 @@ private static void assertDockerInspect(String imageReference)
 
   private ImmutableList<LayerConfiguration> fakeLayerConfigurations;
 
-  private Map<BuildStepType, Integer> layerCounts = new HashMap<>();
+  private final Map<BuildStepType, Integer> layerCounts = new HashMap<>();
   private final Consumer<LayerCountEvent> layerCountConsumer =
       layerCountEvent -> {
-        BuildStepType stepType = layerCountEvent.getBuildStepType();
-        if (layerCounts.containsKey(stepType)) {
-          layerCounts.put(stepType, layerCounts.get(stepType) + layerCountEvent.getCount());
-        } else {
-          layerCounts.put(stepType, layerCountEvent.getCount());
+        synchronized (layerCounts) {
```

Comment:
I would recommend just using a `ConcurrentHashMap` since it tends to interfere less with the parallelism of the build steps.